### PR TITLE
Register binary and text frame handler in websocket as needed & Allow send websocket frame from any verticle in cluster 

### PR DESCRIPTION
**Pull request by dinhnn**
Register binary and text handler as needed. It will reduce number of event bus address
Allow send websocket frame from any verticle in cluster via eventbus

### CODE REVIEW

**Comment by dinhnn** [comment]
I already had an Eclipse account and I thought my CLA was still valid. But it my pull request still not pass Eclipse validation

File: src/main/java/io/vertx/core/http/impl/WebSocketImplBase.java [hunk 1]
```diff
@@ -56,25 +57,37 @@
...
+    this.eventBus = vertx.eventBus();
     this.maxWebSocketFrameSize = maxWebSocketFrameSize;
   }
-
   public String binaryHandlerID() {
+	  return binaryHandlerID(false);
+  }
+  
+  public synchronized String binaryHandlerID(boolean cluster) {
+    if(binaryHandlerID == null){
```
Comment (purplefox) [inline src/main/java/io/vertx/core/http/impl/WebSocketImplBase.java]: cosmetics, please use appropriate whitespace (see code style guidelines on the wiki)

File: src/main/java/io/vertx/core/http/impl/WebSocketImplBase.java [hunk 2]
```diff
@@ -56,25 +57,37 @@
...
+      this.binaryHandlerID = UUID.randomUUID().toString();
+      Handler<Message<Buffer>> binaryHandler = msg -> writeBinaryFrameInternal(msg.body());
+      binaryHandlerRegistration =(cluster?eventBus.<Buffer>consumer(binaryHandlerID):eventBus.<Buffer>localConsumer(binaryHandlerID)).handler(binaryHandler);
+    }
     return binaryHandlerID;
   }
-
+  
   public String textHandlerID() {
+	    return textHandlerID(false);
```
Comment (purplefox) [inline src/main/java/io/vertx/core/http/impl/WebSocketImplBase.java]: indentation

**Comment by purplefox** [comment]
Thanks!

Looks good, a couple of issues:

1. Code style guidelines  need to be followed. In particular we use 2 spaces for indentation not 4 (see the rest of the code), and we use spaces in expressions. You should be able to infer them by looking at the rest of the code and seeing how your code differs.

2. No tests. We would need some tests.

3. Failing Eclipse validation. You would need to sign the Eclipse CLA and sign the commits as explained on the wiki.

**Comment by purplefox** [comment]
If you are having issues with your CLA please re-open this:

https://bugs.eclipse.org/bugs/show_bug.cgi?id=468552

and add a comment and the Eclipse web team will take a look.

**Comment by purplefox** [comment]
basically the Eclipse CLA checker thingy is flaky as hell, it's a PITA :(

**Comment by purplefox** [comment]
@dinhnn hi, any updates on this?

**Comment by eclipsewebmaster** [comment]
What we're really having a problem with are those commits by "dinhnn@dsma.local"

**Comment by purplefox** [comment]
I'm going to close this due to lack of response.

**Pull request closed without merging**